android/app/src/test/java/com/microsoft/codepush/common/utils/FinalizeResourcesTest.java

```diff
@@ -0,0 +1,88 @@
+package com.microsoft.codepush.common.utils;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+import org.junit.runners.Parameterized.Parameter;
+import org.junit.runners.Parameterized.Parameters;
+
+import java.io.Closeable;
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.List;
+
+import static junit.framework.Assert.assertEquals;
+
+/**
+ * This class tests scenarios of {@link FileUtils#finalizeResources} usage.
+ */
+@RunWith(Parameterized.class)
+public class FinalizeResourcesTest {
+
+    /**
+     * {@link FileUtils} instance.
+     */
+    private FileUtils mFileUtils;
+
+    /**
+     * Helper which creates {@link Closeable} resource that doesn't throw {@link IOException} during the close.
+     * @return {@link Closeable} resource
+     */
+    private static Closeable createGoodResource() {
+        return new Closeable() {
+            @Override
+            public void close() throws IOException {
+            }
+        };
+    }
+
+    /**
+     * Helper which creates {@link Closeable} resource that does throw {@link IOException} during the close.
+     * @return {@link Closeable} resource
+     */
+    private static Closeable createBrokenResource(final IOException exceptionToThrow) {
+        return new Closeable() {
+            @Override
+            public void close() throws IOException {
+                if (exceptionToThrow == null) {
+                    throw new IOException();
+                } else {
+                    throw exceptionToThrow;
+                }
+            }
+        };
+    }
+
+    @Parameters(name = "{index}: finalizeResources({0}, null) = {1}")
+    public static Collection<Object[]> data() {
+
+        IOException exception1 = new IOException();
+        IOException exception2 = new IOException();
+
+        return Arrays.asList(new Object[][]{
+                {Arrays.asList(createGoodResource()), null},
```

Comment:
Can't we just use `mock(InputStream.class)` instead of the `createGoodResource`?

Reply:
@annakocheshkova yes we can, but it just worsen understanding test scenario.

Reply:
@ruslan-bikkinin My opinion is use mock where no real object needed. Instead it turns out that you are creating more complex logic than needed.

Reply:
@annakocheshkova Well I believe we need the third one to decide. @sergey-akhalkov what do you think on this?

Reply:
@ruslan-bikkinin, I would prefer to use `mock(InputStream.class)` instead of `createGoodResource `